spec/ddtrace/contrib/ethon/integration_context.rb

```diff
@@ -18,7 +18,7 @@
     )
 
     server.mount_proc '/' do |req, res|
-      sleep(1) if req.query['simulate_timeout']
+      sleep(0.001) if req.query['simulate_timeout']
```

Comment:
It occurred to me that we could do this without the sleeps by using a `Queue` or similar primitive, something like: `timeout_queue.pop if req.query['simulate_timeout']` and then the test would perform the request, and have an `after { timeout_queue << :finish }` to let the webserver thread go.

Reply:
Yeah, we could :)